syntax = "proto3";

package proto;

enum AccessType {
	UNKNOWN = 0;
	READ = 1;
	WRITE = 2;
	UPDATE = 3;
	DELETE = 4;
}

message AuditInfo {
	
	string audit_id = 1;
	string user_id = 2;
	AccessType access_type = 3;
}

message FileAuditRequest{
	     
	string file_id = 1;
	AuditInfo audit_info = 2;
	string signature = 3;    // To verify the signature
	string public_key = 4;   // To verify the signature
}


